swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt

```diff
@@ -168,21 +159,14 @@ abstract class ScheduledAgent(
 
   companion object {
     fun timeToWork(schedule: Schedule, clock: Clock): Boolean {
-      if (!schedule.enabled) {
-        return true
-      }
-
-      val startTime: LocalTime = schedule.getResolvedStartTime()
-      val endTime: LocalTime = schedule.getResolvedEndTime()
-      val now = LocalTime.from(clock.instant().atZone(schedule.getZoneId()))
-
-      return if (startTime <= endTime) {
-        // Work during the day
-        now.isAfter(startTime) && now.isBefore(endTime)
-      } else {
-        // Work during the night
-        now.isAfter(startTime) || now.isBefore(endTime)
-      }
+      return if (schedule.enabled && LocalDate.now(clock).dayOfWeek in schedule.allowedDaysOfWeek) {
```

Comment:
I personally don't like lifting the return out of the if statement because I feel like it pulls us closer to the groovy "oh the last line is implicitly the return of the function, and I find that hard to easily parse when reading code. I prefer to keep the two returns.

Reply:
Think it's different than the groovy last line is the return thing. But for readability, I can see the traditional return idiom to be clearer.